lib/m3u8-parser.js

```diff
@@ -30,17 +55,35 @@ var m3u8parser = module.exports = function() {
   this._write = function(chunk, encoding, callback) {
     var lines = chunk.toString('utf8').split('\n');
     if (lastLine) { lines[0] = lastLine + lines[0]; }
-    lines.forEach(function(line, i) {
-      if (i < lines.length - 1) {
-        parseLine(line);
-      } else {
-        // Save the last line in case it has been broken up.
-        lastLine = line;
+    if (this.isYtdlCore) {
```

Comment:
Add a check for if you want the old functionality or the YouTube functionality. Might want to rename the variable, as I'm sure this isn't _just_ used in YtdlCore.